x/evm/spec/01_concepts.md

```diff
@@ -0,0 +1,207 @@
+    * Stack and Memory: compute state changes
+* Access to account storage (persistent)
+
+### State Transitions with Smart Contracts
+
+Typically smart contracts expose a public ABI,
+which is a list of supported ways a user can interact with a contract.
+To interact with a contract and invoke a state transition,
+a user will submit a tx carrying any amount of gas and a data payload formatted according to the ABI,
+specifying the type of interaction and any additional parameters.
+When the tx is received, the EVM executes the smart contracts' EVM bytecode using the tx payload.
+
+### Executing EVM bytecode
+
+A contract's EVM bytecode consists of basic operations (add, multiply, store, etc...), called **Opcodes**.
+Each Opcode execution requires gas that needs to be paid with the tx.
+The EVM is therefore considered quasi-turing complete,
+as it allows any arbitrary computation,
+but the amount of computations during a contract execution is limited to the amount of gas provided in the tx.
+Each Opcode's [**gas cost**](https://www.evm.codes/) reflects the cost of running these operations on actual computer hardware
+(e.g. `ADD = 3gas` and `SSTORE = 100gas`).
+To calculate the gas consumption of a tx, the gas cost is multiplied by the **gas price**,
+which can change depending on the demand of the network at the time.
+If the network is under heavy load, you might have to pay a higher gas price to get your tx executed.
+If the gas limit is hit (out of gas exception) no changes to the Ethereum state are applied,
+except that the sender's nonce increments and their balance goes down to pay for wasting the EVM's time.
+
+Smart contracts can also call other smart contracts.
+Each call to a new contract creates a new instance of the EVM (including a new stack and memory).
+Each call passes the sandbox state to the next EVM.
+If the gas runs out, all state changes are discarded.
+Otherwise, they are kept.
+
+For further reading, please refer to:
+
+* [EVM](https://eth.wiki/concepts/evm/evm)
+* [EVM Architecture](https://cypherpunks-core.github.io/ethereumbook/13evm.html#evm_architecture)
+* [What is Ethereum](https://ethdocs.org/en/latest/introduction/what-is-ethereum.html#what-is-ethereum)
+* [Opcodes](https://www.ethervm.io/)
+
+## Ethermint as Geth implementation
+
+Ethermint is an implementation of the [Ethereum protocol in Golang](https://geth.ethereum.org/docs/getting-started) (Geth)
+as a Cosmos SDK module.
+Geth includes an implementation of the EVM to compute state transitions.
+Have a look at the [go-ethereum source code](https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go)
+to see how the EVM opcodes are implemented.
+Just as Geth can be run as an Ethereum node,
+Ethermint can be run as a node to compute state transitions with the EVM.
+Ethermint supports Geth's standard [Ethereum JSON-RPC APIs](https://docs.evmos.org/developers/json-rpc/endpoints.html)
+in order to be Web3 and EVM compatible.
```

Comment:
Should this be rewritten to fit Evmos instead of Ethermint?

Reply:
yes, let's rewrite it